helm/templates/deployment.yaml

```diff
@@ -11,6 +11,10 @@ spec:
     matchLabels:
       app.kubernetes.io/name: {{ include "helm.name" . }}
       app.kubernetes.io/instance: {{ .Chart.Name }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
```

Comment:
Why do we need to allow `1` unavailable? If we do so, what happens if there is only 1 replica?
Couldn't we set this to `0` and have it spinned up before anything is spinned down?